src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java

```diff
@@ -238,98 +274,135 @@ private synchronized void syncAcl() {
 		logger.debug("Syncing the ACL to ElasticSearch");
 		try {
 			logger.debug("Loading SearchGuard ACL...");
-			final SearchGuardACL acl = loadAcl(esClient);
+
+			SearchGuardRoles roles = readRolesACL(esClient);
+			SearchGuardRolesMapping rolesMapping = readRolesMappingACL(esClient);
+			
 			logger.debug("Syncing from cache to ACL...");
-			acl.syncFrom(cache, userProfilePrefix);
-			write(esClient, acl);
+			roles.syncFrom(cache, userProfilePrefix);
+			rolesMapping.syncFrom(cache, userProfilePrefix);
+			
+			writeACL(esClient, roles, rolesMapping);
 		} catch (Exception e) {
 			logger.error("Exception while syncing ACL with cache", e);
 		}
 	}
 	
-	private SearchGuardACL loadAcl(Client esClient) throws IOException {
-		GetRequest request = esClient.prepareGet(searchGuardIndex, SEARCHGUARD_TYPE, SEARCHGUARD_ID)
+	// This will be the new way to do this?
+	private SearchGuardRoles readRolesACL(Client esClient) throws IOException {
+		
+		GetRequest getRequest = esClient.prepareGet("searchguard", "roles", "0")
 				.setRefresh(true).request();
-		request.putInContext(OS_ES_REQ_ID, ACL_FILTER_ID);
-		GetResponse response = esClient.get(request).actionGet(); // need to worry about timeout?
-		return mapper.readValue(response.getSourceAsBytes(), SearchGuardACL.class);
+		GetResponse response = esClient.get(getRequest).actionGet(); // need to worry about timeout?
+		
+		SearchGuardRoles roles = new SearchGuardRoles().load(response.getSource());
+		logger.debug("Read in roles '{}'", roles);
+		return roles;
 	}
 	
+	private SearchGuardRolesMapping readRolesMappingACL(Client esClient) throws IOException {
+		
+		GetRequest getRequest = esClient.prepareGet("searchguard", "rolesmapping", "0")
```

Comment:
I'll update them to be constants.  I can see them changing only if Searchguard changed where they look for them; so we should be safe for a while